src/log.h

```diff
@@ -1,73 +1,211 @@
+#define D_WEB_CLIENT_ACCESS 0x00000080 ///< Debug web client access.
+#define D_TC_LOOP           0x00000100 ///< Debug `tc` plugin loop
+#define D_DEFLATE           0x00000200 ///< Debug web client deflate.
+#define D_CONFIG            0x00000400 ///< Debug applicaton configuration API.
+#define D_PLUGINSD          0x00000800 ///< Debug external plugin daemon.
+#define D_CHILDS            0x00001000 ///< Debug creation of child processes.
+#define D_EXIT              0x00002000 ///< Debug application exit.
+#define D_CHECKS            0x00004000 ///< \deprecated Currently not used. 
+#define D_NFACCT_LOOP       0x00008000 ///< Debug NFACCT plugin loop.
+#define D_PROCFILE          0x00010000 ///< Debug parsing of procfiles.
+#define D_RRD_CALLS         0x00020000 ///< Debug round robin database API.
+#define D_DICTIONARY        0x00040000 ///< Debug usage of dictionary API.
+#define D_MEMORY            0x00080000 ///< \deprecated Currently not used.  
+#define D_CGROUP            0x00100000 ///< Debug cgroups collection.
+#define D_REGISTRY          0x00200000 ///< Debug Registry usage.
+#define D_VARIABLES         0x00400000 ///< Debug health variables. @see D_HEALTH
+#define D_HEALTH            0x00800000 ///< Debug health monitoring. @see D_VARIABLES
+#define D_CONNECT_TO        0x01000000 ///< Debug socket connections.
+#define D_SYSTEM            0x80000000 ///< Debug daemon and system.
 
 //#define DEBUG (D_WEB_CLIENT_ACCESS|D_LISTENER|D_RRD_STATS)
 //#define DEBUG 0xffffffff
-#define DEBUG (0)
+#define DEBUG (0)  ///< Default debugging. Debug nothing.
 
-extern unsigned long long debug_flags;
+extern unsigned long long debug_flags; ///< Global debugging setting (D_*).
 
-extern const char *program_name;
+extern const char *program_name; ///< Name of the program
 
-extern int stdaccess_fd;
-extern FILE *stdaccess;
+extern int stdaccess_fd; ///< File descriptor of `stdaccess_filename`.
+extern FILE *stdaccess;  ///< FILE `stdaccess_filename`.
 
-extern const char *stdaccess_filename;
-extern const char *stderr_filename;
-extern const char *stdout_filename;
+extern const char *stdaccess_filename; ///< Filename to log level _access_.
+extern const char *stderr_filename;    ///< Filename to log level _info_, _error_ and _fatal_.
+extern const char *stdout_filename;    ///< Filename to log level _debug_.
 
-extern int access_log_syslog;
-extern int error_log_syslog;
-extern int output_log_syslog;
+extern int access_log_syslog; ///< boolean enables additional logging level _access_ to syslog.
+extern int error_log_syslog;  ///< boolean enables additional logging levels _info_, _error_ and fatal_ to syslog.
+extern int output_log_syslog; ///< boolean enables additional logging level _debug_ to syslog.
 
-extern time_t error_log_throttle_period, error_log_throttle_period_backup;
-extern unsigned long error_log_errors_per_period;
+extern time_t error_log_throttle_period;          ///< ktsaou: Your help needed.
```

Comment:
`error_log_throttle_period` gives the time in seconds the log should be suspended. Log throttling is not rolling. So, if `error_log_throttle_period = 3600`, the log throttling variables are reset once every hour.

`error_log_throttle_period_backup` is used to restore throttling in case netdata suspends this functionality and needs to resume it. Error throttling is suspended when saving the database (so, you will always see these errors).

`error_log_errors_per_period` this is the threshold. If that many errors are logged, error logging is suspended for the end of the period.